sdk/src/Services/S3/Custom/AmazonS3Config.cs

```diff
@@ -35,13 +35,14 @@ public partial class AmazonS3Config : ClientConfig
         private const string DefaultProfileName = "default";
         private const string AwsS3UsEast1RegionalEndpointsEnvironmentVariable = "AWS_S3_US_EAST_1_REGIONAL_ENDPOINT";
         private const string DisableMRAPEnvName = "AWS_S3_DISABLE_MULTIREGION_ACCESS_POINTS";
+        private const string AwsConfigFileEnvName = "AWS_CONFIG_FILE";
 
         private bool forcePathStyle = false;
         private bool useAccelerateEndpoint = false;
         private S3UsEast1RegionalEndpointValue? s3UsEast1RegionalEndpointValue;
         private readonly string legacyUSEast1GlobalRegionSystemName = RegionEndpoint.USEast1.SystemName;
 
-        private static CredentialProfileStoreChain credentialProfileChain = new CredentialProfileStoreChain();
+        private static CredentialProfileStoreChain credentialProfileChain = new CredentialProfileStoreChain(Environment.GetEnvironmentVariable(AwsConfigFileEnvName));
```

Comment:
I'm open to leaving this as is. We added support for `AWS_CONFIG_FILE` a couple years ago and when this class was created, this environment variable wasn't available. Now that it is, it makes sense for the `CredentialProfileStoreChain` to check this value to see if it should look somewhere else for the config file. 

Given that the constructor accepts a `ProfileLocation`, I thought it made sense to add it here. If it isn't set, then it will just continue to work as it did previously.

Reply:
Okay, given there are other services that need the update, we can do it all in a separate pr👍